src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,253 @@
+package Duke;
+import Duke.exceptions.IllegalDate;
+import Duke.exceptions.IllegalIndex;
+import Duke.exceptions.IllegalDescription;
+
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+
+public class Duke {
+    protected static ArrayList<Task> list = new ArrayList<>();
+    protected static int numberOfTasks = 0;
+    public static void main(String[] args) throws IOException {
+        Save save = new Save();
+        String home = System.getProperty("user.home");
+        //Read from memory
+        save.readFile(home+"/Documents/log.txt");
+        boolean hasExit = false;
+
+        // Problem will come if there are more than 100 tasks
+//        Task[] list = new Task[100];
+        printWelcomeMessage();
+        while (!hasExit) {
+            // Read User input
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+
+            // process user input and decide what operation to use
+            String operation = extractOperationType(line);
+            String printStatement;
+
+            switch (operation){
+                case "bye":
```

Comment:
I think it would be better to express the case conditions like "list" "bye" "done" etc. as constants.

Reply:
planning to do so in the next commit..